aws-iot-securityprofile/src/main/java/com/amazonaws/iot/securityprofile/Translator.java

```diff
@@ -0,0 +1,326 @@
+        if (iotCriteria == null) {
+            return null;
+        }
+        return BehaviorCriteria.builder()
+                .comparisonOperator(iotCriteria.comparisonOperatorAsString())
+                .value(translateMetricValueFromIotToCfn(iotCriteria.value()))
+                .durationSeconds(iotCriteria.durationSeconds())
+                .consecutiveDatapointsToAlarm(iotCriteria.consecutiveDatapointsToAlarm())
+                .consecutiveDatapointsToClear(iotCriteria.consecutiveDatapointsToClear())
+                .statisticalThreshold(translateStatisticalThresholdFromIotToCfn(iotCriteria.statisticalThreshold()))
+                .build();
+    }
+
+    static software.amazon.awssdk.services.iot.model.MetricValue translateMetricValueFromCfnToIot(
+            MetricValue cfnMetricValue) {
+        if (cfnMetricValue == null) {
+            return null;
+        }
+        // The MetricValue is a String in the CFN model because the json-schema doesn't support Long numbers.
+        // This is a known issue and some existing services are already using Strings as the workaround.
+        // For example, AWS::ElasticLoadBalancing::LoadBalancer HealthCheck
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-elb-health-check.html
+        String countStringFromCfn = cfnMetricValue.getCount();
+        Long countLong;
+        if (countStringFromCfn == null) {
+            countLong = null;
+        } else {
+            try {
+                countLong = Long.parseLong(countStringFromCfn);
+            } catch (NumberFormatException e) {
+                throw new CfnInvalidRequestException("Invalid value for Behavior::MetricValue::Count: " +
+                                                     countStringFromCfn);
+            }
+        }
+
+        return software.amazon.awssdk.services.iot.model.MetricValue.builder()
+                .count(countLong)
+                .cidrs(cfnMetricValue.getCidrs())
+                .ports(cfnMetricValue.getPorts())
+                .build();
+    }
+
+    static MetricValue translateMetricValueFromIotToCfn(
+            software.amazon.awssdk.services.iot.model.MetricValue iotMetricValue) {
+        if (iotMetricValue == null) {
+            return null;
+        }
+
+        MetricValue.MetricValueBuilder metricValueBuilder = MetricValue.builder();
+        if (iotMetricValue.count() != null) {
+            metricValueBuilder.count(iotMetricValue.count().toString());
+        }
+        // For lists, we're using the .has* methods to differentiate between null and empty lists
+        // from DescribeSecurityProfileResponse.
+        // SDK converts nulls from Describe API to empty DefaultSdkAutoConstructLists,
+        // so if we simply translate without the .has* check, nulls will turn into empty lists.
+        if (iotMetricValue.hasCidrs()) {
+            metricValueBuilder.cidrs(new HashSet<>(iotMetricValue.cidrs()));
```

Comment:
Shouldn't this be Arrays, since we defining the type as Arrays.

Reply:
(Discussed offline) We've modeled Cidrs as well as other lists like Behaviors to have "insertionOrder=false", because the order of the elements doesn't matter to the service. That's why the generated Java objects have Sets not Lists.